lib/actions/adapterActions.js

```diff
@@ -1097,6 +1109,42 @@ function closeSelectedAdapter(dispatch, getState) {
     });
 }
 
+export function selectedSerialPort(port) {
+    return dispatch => (
+        new Promise(() => {
+            dispatch(closeAdapter(() => {
+                getDeviceInfo(port.serialNumber)
+                    .then(deviceInfo => {
+                        dispatch(adapterGetLocalDeviceInfoAction(deviceInfo));
+                        logger.info(`Current device family is
+                            ${familyDefinitions[deviceInfo.family].split('_')[0].replace('NRF', 'nRF')}
+                            (${('0x0' + (deviceInfo.family).toString(16)).substring(-2)})`);
+                        logger.info(`Current device type is
+                            ${deviceTypeDefinitions[deviceInfo.deviceType].split('_')[0].replace('NRF', 'nRF')}
+                            (${('0x0' + (deviceInfo.deviceType).toString(16)).substring(-2)})`);
```

Comment:
Similar to the above, I think this can be simplified by actually putting the values we want to show right in the `deviceTypeDefinitions`. No need for all the string operations. Also consider if pc-nrfjprog-js should provide a string representation of the device type.